tests/appsec/waf/test_telemetry.py

```diff
@@ -0,0 +1,88 @@
+from utils import interfaces, released, rfc, weblog, coverage, scenario, context
+from utils.tools import logger
+
+TELEMETRY_REQUEST_TYPE_GENERATE_METRICS = "generate-metrics"
+
+
+def _validate_headers(headers, language):
+    """https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/how-to-use.md
+    """
+    expected_headers = {
+        "Content-type": "application/json",
+        "DD-Telemetry-API-Version": "v1",
+        "DD-Telemetry-Request-Type": TELEMETRY_REQUEST_TYPE_GENERATE_METRICS,
+        "DD-Client-Library-Language": language,
+        "DD-Client-Library-Version": "",
+        "DD-Agent-Env": "",
+        "DD-Agent-Hostname": "",
+    }
+
+    for key, value in headers:
+        if expected_headers.get(key) is not None:
+            expected_value = expected_headers.pop(key)
+            if expected_value != "":
+                assert value == expected_value
+            else:
+                assert value is not None, f"Empty `{key}` header"
+
+    if len(expected_headers) > 0:
+        raise AssertionError(
+            "Headers %s not found in payload headers: %s" % ([header for header in expected_headers], headers)
+        )
+
+
+def _validate_metrics(payload, language):
```

Comment:
`language` is always equals to `context.library`

```suggestion
def _validate_metrics(payload):
```